Utilities/src/test/java/edu/stanford/hivdb/mutations/MutationTest.java

```diff
@@ -85,13 +85,21 @@ public void testNormalizeAAs() {
 		assertEquals("_", Mutation.normalizeAAs("Insertion"));
 		assertEquals("-", Mutation.normalizeAAs("Deletion"));
 		assertEquals("ACDE", Mutation.normalizeAAs("DECA"));
-		assertEquals("ACDE", Mutation.normalizeAAs("deca"));
+		assertEquals("-ACE", Mutation.normalizeAAs("deca"));
 	}
 	
-//	@Test
-//	public void testExtractGene() {
-//		TODO: Refactor class to reduce redundancy between normalization logic. 
-//	}
+	@Test
+	public void testExtractGene() {
+		assertEquals(Gene.PR, Mutation.extractGene("PR100A"));
+		assertEquals(Gene.IN, Mutation.extractGene("IN100A"));
+		assertEquals(Gene.RT, Mutation.extractGene("RT100A"));
+		assertEquals(null, Mutation.extractGene("not a mutation"));
+	}
+	
+	@Test(expected=InvalidMutationStringException.class)
+	public void testExtractGeneWithMalformedMut() {
+		assertEquals(null, Mutation.extractGene("P100D"));
```

Comment:
Can you explain why `P100D` invalid again? I'm confused.

Reply:
While P100D is a valid mutation string, it is invalid in the context of `extractGene()`. Specifically, `extractGene()`, chooses to throw an `InvalidMutationStringException` if `Gene.valueOf(m.group(1).toUpperCase())` fails. 

Indeed, throwing this specific exception is a bit misleading.

Reply:
Oo good tip.